.github/workflows/lint.yml

```diff
@@ -24,3 +24,22 @@ jobs:
         with:
           python-version: "3.x"
       - uses: pre-commit/action@v3.0.0
+
+  patchcheck:
+    if: "github.repository == 'python/cpython' && github.event_name == 'pull_request'"
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
```

Comment:
Can we avoid checking out the full 33 years and 120k commit history?

CAM originally had the same in https://github.com/python/cpython/pull/108065 but figured out a way around it.